lib/IRGen/IRGenDebugInfo.cpp

```diff
@@ -2330,9 +2337,61 @@ void IRGenDebugInfoImpl::emitArtificialFunction(IRBuilder &Builder,
   setCurrentLoc(Builder, Scope, ALoc);
 }
 
+void IRGenDebugInfoImpl::handleFragmentDIExpr(
+    SILDebugInfoExpression::const_iterator &It,
+    SmallVectorImpl<uint64_t> &Operands) {
+  const SILDIExprOperand &CurOp = *It++;
+  assert(CurOp.getAsOperator() == SILDIExprOperator::Fragment);
+  // Expecting a VarDecl that points to a field in an struct
+  auto *VD = dyn_cast_or_null<VarDecl>((It++)->getAsDecl());
```

Comment:
In DIExpression, we define a custom iterator to decode the uint64_t array. Maybe we want to copy this?

Reply:
I'm not sure I understand you correctly: SILDebugInfoExpression doesn't store a list of uin64_t but some high-level objects like `Decl*`, so I don't think we can reuse llvm::DIExpression's iterator here.

Reply:
Would it make sense to also define a (new) iterator _similar_ to DIExpression's that makes it convenient to iterate through the list of SILDIExprOperand one operator at a time. So operator++ would skip over the entire Fragment, including its two parameters.

Reply:
That is actually a good idea! Will add this feature